### PR TITLE
fix: Resolve linting issues

### PR DESCRIPTION
**Pull request by rmi22186**
## Instructions
 1. PR target branch should be against `development`
 2. PR title name should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-title-check.yml
 3. PR branch prefix should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-branch-check-name.yml

 ## Summary
Linter showed that there were several `Unchecked cast` errors.  This PR resolves them.

 ## Testing Plan
 - [X] Was this tested locally? If not, explain why.
I manually tested events that included a product, impression, promotion event with custom attributes, as well as added CCPA to ensure all data was still sent properly with no crashes. 

 ## Reference Issue (For mParticle employees only.  Ignore if you are an outside contributor)
 - Closes https://go.mparticle.com/work/SQDSDKS-6929

### CODE REVIEW

**Review by Mansi-mParticle — COMMENTED** [review]
Other code is looking good to me only one comment I have

File: android/src/main/kotlin/com/mparticle/mparticle_flutter_sdk/MparticleFlutterSdkPlugin.kt [hunk 1]
```diff
@@ -750,20 +800,25 @@ class MparticleFlutterSdkPlugin: FlutterPlugin, MethodCallHandler {
...
+              .build()
     }
+            ?: run {
+              result.error(TAG, "Missing \"consented\" value for arguments: $arguments", null)
+              null
+            }
+  } catch (ex: Exception) {
+    result.error(TAG, ex.message, null)
+    null
   }
```
[the remark this answers is not in the document]
Comment (rmi22186) [reply android/src/main/kotlin/com/mparticle/mparticle_flutter_sdk/MparticleFlutterSdkPlugin.kt:819]: actually this is just a line formatting issue. i have fixed this.  the builds from CI are all working, showing this isn't a syntax issue.

File: android/src/main/kotlin/com/mparticle/mparticle_flutter_sdk/MparticleFlutterSdkPlugin.kt [hunk 2]
```diff
@@ -751,20 +801,24 @@ class MparticleFlutterSdkPlugin: FlutterPlugin, MethodCallHandler {
...
+          result.error(TAG, "Missing \"consented\" value for arguments: $arguments", null)
+          null
+      }
     } catch (ex: Exception) {
       result.error(TAG, ex.message, null)
       null
     }
   }
 
+
```
Comment (Mansi-mParticle) [inline android/src/main/kotlin/com/mparticle/mparticle_flutter_sdk/MparticleFlutterSdkPlugin.kt:821]:
```suggestion
```

**Review by Mansi-mParticle — COMMENTED** [review]
Remove Extra line

**Pull request merged**